ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvm/src/io/ktor/server/plugins/ContentNegotiation.kt

```diff
@@ -187,13 +182,13 @@ public class ContentNegotiation internal constructor(
                 }
 
                 val rendered = converted?.let { transformDefaultContent(it) }
-                    ?: HttpStatusCodeContent(HttpStatusCode.NotAcceptable)
+                    ?: return@intercept
 
                 val contentType = rendered.contentType
                 if (plugin.checkAcceptHeader(acceptItems, contentType)) {
                     proceedWith(rendered)
                 } else {
-                    proceedWith(HttpStatusCodeContent(HttpStatusCode.NotAcceptable))
+                    return@intercept
```

Comment:
In this case, the converter did work and returned invalid result. It needs to be checked here. Please revert this change.